sql-plugin/src/main/scala/com/nvidia/spark/rapids/RapidsDiskStore.scala

```diff
@@ -45,58 +91,44 @@ class RapidsDiskStore(diskBlockManager: RapidsDiskBlockManager)
         case h: HostMemoryBuffer => h
         case _ => throw new UnsupportedOperationException("buffer without host memory")
       }
-      val id = incoming.id
-      val path = if (id.canShareDiskPaths) {
-        sharedBufferFiles.computeIfAbsent(id, _ => id.getDiskPath(diskBlockManager))
-      } else {
-        id.getDiskPath(diskBlockManager)
-      }
-      val fileOffset = if (id.canShareDiskPaths) {
-        // only one writer at a time for now when using shared files
-        path.synchronized {
-          copyBufferToPath(hostBuffer, path, append = true)
+      val iter = new HostByteBufferIterator(hostBuffer)
+      val fos = new FileOutputStream(path, append)
+      try {
+        val channel = fos.getChannel
+        val fileOffset = channel.position
+        iter.foreach { bb =>
```

Comment:
These `ByteBuffer` instances are views on top of `hostBuffer`. It looks like the jdk will instantiate a `DirectByteBuffer` with a null cleaner in this case. We call `env->NewDirectByteBuffer(addr, len)`. 

As such I think these views are going to stay around in the heap as it is. We could move this implementation to cuDF so we can write directly from the `HostMemoryBuffer` to a file directly, rather than having to work around limitations in the jdk's `ByteBuffer` impl with this iterator.

Reply:
I know that closing them really ends up being a NOOP and the heap cleans them up. The only reason I mention it, is because if `bb` changes in some way to not be backed by a HostMemoryBuffer we would then potentially leak memory. It is just defensive. And a nit.